include/multipass/process.h

```diff
@@ -51,16 +71,14 @@ class Process : public QObject
 
     virtual qint64 write(const QByteArray& data) = 0;
 
-    virtual bool run_and_return_status(const int timeout = 30000) = 0;
-    virtual QString run_and_return_output(const int timeout = 30000) = 0;
+    virtual const ProcessExitState run_and_return_exit_state(const int timeout = 30000) = 0;
```

Comment:
1. Yeah name isn't the most elegant, I'm happy to change it. Does `run` suggest it it supposed to run to completion? I consider it a synonym to `start`. But as long as its clear to us...
2. Yeah I'll make it a const ref, it started out as a small struct where a copy would have been cheaper.
Thanks for the feedback

Reply:
Actually I want a const value as a const ref will point to a local that will be out of scope.

Reply:
1. I would be ok with `run`, but I wouldn't mind something like, `run_to_completion`, `run_and_wait`, `blocking_run`, or similar.
2. I am perfectly fine with the return-by-value really, it is the const that confused me: why impose const on the client if it is his own copy? I thought maybe it had started as a const-ref or something. It can't be a reference now though, otherwise linux_process implementation would be returning a reference to a local.

Reply:
Just realized the messages crossed. Yeah, my point is basically I don't think the const makes sense without the reference. It is basically restricting our post condition arbitrarily, as it does not influence `Process` anymore at that point. But I won't block on that...

Reply:
FWIW, a couple of references about returning const values:
- https://stackoverflow.com/questions/8716330/purpose-of-returning-by-const-value 
- https://stackoverflow.com/questions/6299967/what-are-the-use-cases-for-having-a-function-return-by-const-value-for-non-built

Reply:
I see no reason why ProcessExitState should be mutable for the consumer in this case. If it causes a future impediment this const can be dropped.

Reply:
I've renamed it to "execute", as it matches more closely with Qt's naming.